src/Elmish.WPF/Binding.fs

```diff
@@ -1684,6 +2148,35 @@ type Binding private () =
       canExec
       (defaultArg uiBoundCmdParam false)
 
+  /// <summary>
+  ///   Creates a <c>Command</c> binding that depends on the
+  ///   <c>CommandParameter</c>
+  ///   and can execute if <paramref name="canExec" /> returns <c>true</c>.
+  /// </summary>
+  /// <param name="exec">Returns the message to dispatch.</param>
+  /// <param name="canExec">Indicates whether the command can execute.</param>
+  /// <param name="uiBoundCmdParam">
+  ///   If <c>true</c>, <c>CanExecuteChanged</c> will trigger every time WPF's
+  ///   <c>CommandManager</c>
+  ///   detects UI changes that could potentially influence the command's
+  ///   ability to execute. This will likely lead to many more triggers than
+  ///   necessary, but is needed if you have bound the <c>CommandParameter</c>
+  ///   to another UI property.
+  /// </param>
+  /// <param name="wrapDispatch">
+  ///   Wraps the dispatch function with additional behavior, such as
+  ///   throttling, debouncing, or limiting.
+  /// </param>
+  [<System.Obsolete("In version 5, this method will be removed.  Use the overload without the \"wrapDispatch\" parameter followed by a call to \"Binding.alterMsgStream\".  For an example, see how this method is implemented.")>]
+  static member cmdParamIf
```

Comment:
For this one method with two optional parameters, how about creating these three overloads?
1. optional `uiBoundCmdParam` and no `wrapDispatch`
2. no `uiBoundCmdParam` and required `wrapDispatch` with `Obsolete` attribute
3. required `uiBoundCmdParam` and required `wrapDispatch` with `Obsolete` attribute

Two optional parameters means there is four ways to pass in arguments.  These three overloads cover those four cases because the first overload covers two cases and the other two overloads each cover one case.

Does this sound correct to you?